src/core/entity/dbus_query.cpp

```diff
@@ -378,7 +468,11 @@ const DBusPropertiesMap
     DBusInstance::queryProperties(const connect::DBusConnectUni& connect,
                                   const InterfaceName& interface)
 {
-    DBusPropertiesMap properties;
+    // The IInstance might be freed by another thread, keep it hold until
+    // dbus-query processing.
+    // The variable will be unused, only for protection
+    // purposes to keep alive dbus-connection.
+    volatile const auto selfShrGuard = this->shared_from_this();
```

Comment:
Very strange concept.
If someone got shared_ptr it should keep it until usage finishes.
In which case this rule is broken?

Reply:
This is an artifact of my research. I have removed that.